src/components/ha-wa-dialog.ts

```diff
@@ -0,0 +1,380 @@
+        --show-duration: var(--ha-dialog-show-duration, 200ms);
+        --hide-duration: var(--ha-dialog-hide-duration, 200ms);
+        --ha-dialog-surface-background: var(
+          --card-background-color,
+          var(--ha-color-surface-default)
+        );
+        --wa-color-surface-raised: var(
+          --ha-dialog-surface-background,
+          var(--card-background-color, var(--ha-color-surface-default))
+        );
+        --wa-panel-border-radius: var(
+          --ha-dialog-border-radius,
+          var(--ha-border-radius-3xl)
+        );
+        max-width: var(--ha-dialog-max-width, 100vw);
+        max-width: var(--ha-dialog-max-width, 100svw);
+      }
+
+      :host([width="small"]),
+      :host(.size-changed[width-on-title-click="small"]) wa-dialog {
+        --width: var(--ha-dialog-width-sm, min(320px, var(--full-width)));
+      }
+
+      :host([width="large"]),
+      :host(.size-changed[width-on-title-click="large"]) wa-dialog {
+        --width: var(--ha-dialog-width-lg, min(720px, var(--full-width)));
+      }
+
+      :host([width="full"]),
+      :host(.size-changed[width-on-title-click="full"]) wa-dialog {
+        --width: var(--full-width);
+      }
+
+      wa-dialog::part(dialog) {
+        min-width: var(--width, var(--full-width));
+        max-width: var(--width, var(--full-width));
+        max-height: var(
+          --ha-dialog-max-height,
+          calc(100% - var(--ha-space-20))
+        );
+        position: var(--dialog-surface-position, relative);
+        margin-top: var(--dialog-surface-margin-top, auto);
+        transition:
+          min-width var(--ha-dialog-expand-duration, 200ms) ease-in-out,
+          max-width var(--ha-dialog-expand-duration, 200ms) ease-in-out;
+        display: flex;
+        flex-direction: column;
+        overflow: hidden;
+      }
+
+      @media all and (max-width: 450px), all and (max-height: 500px) {
+        :host {
+          --ha-dialog-border-radius: var(--ha-space-0);
+        }
+
+        wa-dialog {
+          --full-width: var(--ha-dialog-width-full, 100vw);
+        }
+
+        wa-dialog::part(dialog) {
+          min-height: var(--ha-dialog-min-height, 100vh);
+          min-height: var(--ha-dialog-min-height, 100svh);
+          max-height: var(--ha-dialog-max-height, 100vh);
+          max-height: var(--ha-dialog-max-height, 100svh);
+          padding-top: var(--safe-area-inset-top, var(--ha-space-0));
+          padding-bottom: var(--safe-area-inset-bottom, var(--ha-space-0));
+          padding-left: var(--safe-area-inset-left, var(--ha-space-0));
+          padding-right: var(--safe-area-inset-right, var(--ha-space-0));
+        }
+      }
+
+      wa-dialog::part(header) {
+        max-width: 100%;
+        overflow: hidden;
+        display: flex;
+        align-items: center;
+        padding: var(--ha-space-6) var(--ha-space-6) var(--ha-space-4)
+          var(--ha-space-6);
+        gap: var(--ha-space-1);
+      }
```

Comment:
Do we need this? We disable the header from them, so the close button should be not rendered.